artifactory/commands/transferfiles/delayedartifactshandler.go

```diff
@@ -167,18 +172,56 @@ func getDelayUploadComparisonFunctions(packageType string) []shouldDelayUpload {
 }
 
 type delayUploadHelper struct {
-	shouldDelayFunctions    []shouldDelayUpload
-	delayedArtifactsChannel chan FileRepresentation
+	shouldDelayFunctions       []shouldDelayUpload
+	delayedArtifactsChannelMng *DelayedArtifactsChannelMng
 }
 
 // Decide whether to delay the deployment of a file by running over the shouldDelayUpload array.
 // When there are multiple levels of requirements in the deployment order, the first comparison function in the array can be removed each time in order to no longer delay by that rule.
-func (delayHelper delayUploadHelper) delayUploadIfNecessary(file FileRepresentation) bool {
+func (delayHelper delayUploadHelper) delayUploadIfNecessary(file FileRepresentation) (delayed, shouldStop bool) {
 	for _, shouldDelay := range delayHelper.shouldDelayFunctions {
 		if shouldDelay(file.Name) {
-			delayHelper.delayedArtifactsChannel <- file
-			return true
+			delayed = true
+			succeed := delayHelper.delayedArtifactsChannelMng.add(file)
+			if !succeed {
+				// In case an error occurred while handling delayed artifacts - stop transferring.
+				log.Debug("Stop transferring data - error occurred while handling transfer's errors files.")
+				shouldStop = true
+			}
 		}
 	}
-	return false
+	return
+}
+
+// DelayedArtifactsChannelMng managing writing 'delayed artifacts' to a common channel.
+// In case that an error occurred while handling the files - stops adding elements to the channel.
+type DelayedArtifactsChannelMng struct {
+	channel chan FileRepresentation
+	err     error
+}
+
+// Check if a new element can be added to the channel
```

Comment:
This description does not accurately describe this function, because this function does more than just check,